repos/system_upgrade/el7toel8/actors/bindupdate/libraries/updates.py

```diff
@@ -0,0 +1,85 @@
+from leapp.libraries.common import isccfg
+from leapp.libraries.stdlib import run, api
+
+# Callback for walk function
+callbacks = {
+    'dnssec-lookaside': isccfg.ModifyState.callback_comment_out,
+}
+
+
+def paths_from_issues(issues):
+    """Extract paths from list of BindConfigIssuesModel."""
+    paths = []
+    for issue in issues:
+        paths.append(issue.path)
+    return paths
+
+
+def parser_file(parser, path):
+    for cfg in parser.FILES_TO_CHECK:
+        if cfg.path == path:
+            return cfg
+    return None
+
+
+def make_backup(path, backup_suffix='.leapp'):
+    """Make backup of a file before modification."""
+    backup_path = path + backup_suffix
+    run(['cp', '--preserve=all', path, backup_path])
+
+
+def update_section(parser, section):
+    """Modify one section.
+
+    :type section: ConfigSection
+    """
+    state = isccfg.ModifyState()
+    parser.walk(section, callbacks, state)
+    state.finish(section)
+    return state.content()
+
+
+def update_config(parser, cfg):
+    """Modify contents of file accoriding to rules.
+
+    :type cfg: ConfigFile
+    :returns str: Modified config contents
+    """
+    return update_section(parser, cfg.root_section())
+
+
+def update_file(parser, path, write=True):
+    """Prepare modified content for the file, make backup and rewrite it.
+
+    :param parser: IscConfigParser
+    :param path: String with path to a file
+    :param log: Log instance with debug(str) method or None
+    :param write: True to allow file modification, false to only return modification status
+    """
+    cfg = parser_file(parser, path)
+    modified = update_config(parser, cfg)
+    if modified != cfg.buffer:
+        api.current_logger().debug('%s needs modification', path)
+        if write:
+            make_backup(path)
+            with open(path, 'w') as f:
+                f.write(modified)
+            api.current_logger().debug('%s updated to size %d', path, len(modified))
+        return True
+    return False
+
+
+def update_facts(facts, path='/etc/named.conf'):
+    """Parse and update all files according to supplied facts.
+
+    :param facts: BindFacts instance
+    :param path: String to main configuration file
+    :returns: number of modified files
+    """
+    parser = isccfg.IscConfigParser(path)
+    modified_files = 0
+    if facts.dnssec_lookaside:
+        for model in facts.dnssec_lookaside:
+            if update_file(parser, model.path):
+                modified_files += 1
+    return modified_files
```

Comment:
Just asking: would it make sense to keep track of which exact files were modified? And store this information in the report?

Reply:
Good idea, I just wanted to log something. But I guess list of modified files should be kept in a single place.